modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/executors/ReactorScheduledExecutorService.java

```diff
@@ -122,6 +125,16 @@ public void execute(Runnable command) {
         delegate.execute(decorateRunnable(command));
     }
 
+    @Override
+    protected <T> RunnableFuture<T> newTaskFor(Runnable runnable, T value) {
+        return new UninterruptibleFuture<>(super.newTaskFor(runnable, value));
+    }
+
+    @Override
+    protected <T> RunnableFuture<T> newTaskFor(Callable<T> callable) {
+        return new UninterruptibleFuture<>(super.newTaskFor(callable));
+    }
```

Comment:
Are these creating a new task for each read operation? Just thinking about how promptly a read might be cancelled without the interrupt.

Reply:
Yes, see `org.elasticsearch.repositories.azure.AzureBlobStore#convertStreamToByteBuffer`. We're reading from the input stream in 64Kb chunks, and we emit those in order; meaning that after the pipeline has been cancelled at most we'll need to wait until we're able to read 64Kb from disk.